### PR TITLE
Improve cli send transfer command

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- supports synthetic warp routes
- fixes bug in collateral balance checking
- deduces token balance checks from provided router address

### Testing

- Manual and CLI CI test

### CODE REVIEW

**Review by aroralanuk — APPROVED** [review]

File: typescript/cli/src/send/transfer.ts [hunk 1]
```diff
@@ -75,31 +74,37 @@ export async function sendTestTransfer({
...
-    );
-    const tokenAddress = await collateralRouter.wrappedToken();
+  // TODO: move to SDK token router app
+  // deduce TokenType
+  // 1. decimals() call implies synthetic
+  // 2. wrappedToken() call implies collateral
+  // 3. if neither, it's native
+  let tokenAddress: Address | undefined;
+  const provider = multiProvider.getProvider(origin);
+  try {
```
Comment (aroralanuk) [inline typescript/cli/src/send/transfer.ts:85]: nit: not a big fan of the nested try catch logic here but I don't know of any other alternatives

File: typescript/cli/src/send/transfer.ts [hunk 2]
```diff
@@ -75,31 +74,37 @@ export async function sendTestTransfer({
...
+      const tokenRouter = HypERC20Collateral__factory.connect(
+        routerAddress,
+        provider,
+      );
+      tokenAddress = await tokenRouter.wrappedToken();
+    } catch (error) {}
+  }
+
+  if (tokenAddress) {
     await assertTokenBalance(
```
Comment (aroralanuk) [inline typescript/cli/src/send/transfer.ts:104]: add a comment suggesting this covers both the synthetic and collateral cases

**Review by aroralanuk — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]
Could be a neater but this can probably use the WarpCore soon

File: typescript/cli/src/send/transfer.ts (hunk 1 above)
Comment (yorhodes) [reply typescript/cli/src/send/transfer.ts:85]: me neither tbh

**Pull request merged**